[mosquitto] Static linking broken

### Description / Steps to reproduce the issue

I try to use the libmosquitto package in my project with static linking. I use `-static` linker flag and I get the linker flags for the package like this:  `pkg-config --libs --static libmosquitto`. The output of the pkg-config is `-lmosquitto`.

### Expected behavior

The flags provided by pkg-config should work as is.

### Actual behavior

The linker fails like this: `cannot find -lmosquitto: No such file or directory`
It happens because the static library is called `libmosquitto.dll.a`, see the file list [here](https://packages.msys2.org/package/mingw-w64-x86_64-mosquitto).
If I link with the `-lmosquitto.dll` flag, the link succeeds.
I think the `libmosquitto.dll.a` should be renamed to `libmosquitto.a`, but I don't know where to look at.
I think `libmosquittopp` is affected too, but I haven't tried it.

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-19045

### MINGW environments affected

- [X] MINGW64
- [ ] MINGW32
- [ ] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

### Are you willing to submit a PR?

_No response_

Unfortunately no, I get various linker errors like these
```
undefined reference to `__imp_mosquitto_lib_init'
undefined reference to `__imp_mosquitto_new'
undefined reference to `__imp_mosquitto_connect_callback_set'
undefined reference to `__imp_mosquitto_disconnect_callback_set'
undefined reference to `__imp_mosquitto_message_callback_set'
undefined reference to `__imp_mosquitto_log_callback_set'
undefined reference to `__imp_mosquitto_connect'
...
```
I didn't expect such a quick response, so I opened a ticket in the mosquitto repo too.
https://github.com/eclipse/mosquitto/issues/3035

I don't understand how could a compiler flag solve a linker issue.
Anyway, I tried it, and the original issue still persists: `cannot find -lmosquitto: No such file or directory`

--- a/mingw-w64-mosquitto/0001-mosquitto-2.0.18-fix-static-libname.patch
+++ b/mingw-w64-mosquitto/0001-mosquitto-2.0.18-fix-static-libname.patch
@@ -1,0 +1,24 @@
+diff -Naur mosquitto-2.0.18.orig/lib/CMakeLists.txt mosquitto-2.0.18/lib/CMakeLists.txt
+--- mosquitto-2.0.18.orig/lib/CMakeLists.txt	2024-04-10 00:32:54.825031800 +0300
++++ mosquitto-2.0.18/lib/CMakeLists.txt	2024-04-10 00:35:12.810000800 +0300
+@@ -112,7 +112,7 @@
+ 	target_link_libraries(libmosquitto_static ${LIBRARIES})
+ 
+ 	set_target_properties(libmosquitto_static PROPERTIES
+-		OUTPUT_NAME mosquitto_static
++		OUTPUT_NAME mosquitto
+ 		VERSION ${VERSION}
+ 	)
+ 
+diff -Naur mosquitto-2.0.18.orig/lib/cpp/CMakeLists.txt mosquitto-2.0.18/lib/cpp/CMakeLists.txt
+--- mosquitto-2.0.18.orig/lib/cpp/CMakeLists.txt	2024-04-10 00:32:54.809404400 +0300
++++ mosquitto-2.0.18/lib/cpp/CMakeLists.txt	2024-04-10 00:36:26.151017700 +0300
+@@ -33,7 +33,7 @@
+ 	target_link_libraries(mosquittopp_static ${LIBRARIES})
+ 
+ 	set_target_properties(mosquittopp_static PROPERTIES
+-		OUTPUT_NAME mosquittopp_static
++		OUTPUT_NAME mosquittopp
+ 		VERSION ${VERSION}
+ 	)
+ 

--- a/mingw-w64-mosquitto/0002-mosquitto-2.0.18-pkgconf-static-cflags.patch
+++ b/mingw-w64-mosquitto/0002-mosquitto-2.0.18-pkgconf-static-cflags.patch
@@ -1,0 +1,18 @@
+diff -Naur mosquitto-2.0.18.orig/libmosquitto.pc.in mosquitto-2.0.18/libmosquitto.pc.in
+--- mosquitto-2.0.18.orig/libmosquitto.pc.in	2024-04-10 00:32:55.137531100 +0300
++++ mosquitto-2.0.18/libmosquitto.pc.in	2024-04-10 21:47:48.600521000 +0300
+@@ -7,4 +7,5 @@
+ Description: mosquitto MQTT library (C bindings)
+ Version: @VERSION@
+ Cflags: -I${includedir}
++Cflags.private: -DLIBMOSQUITTO_STATIC
+ Libs: -L${libdir} -lmosquitto
+diff -Naur mosquitto-2.0.18.orig/libmosquittopp.pc.in mosquitto-2.0.18/libmosquittopp.pc.in
+--- mosquitto-2.0.18.orig/libmosquittopp.pc.in	2024-04-10 00:32:54.809404400 +0300
++++ mosquitto-2.0.18/libmosquittopp.pc.in	2024-04-10 21:47:48.616141900 +0300
+@@ -7,4 +7,5 @@
+ Description: mosquitto MQTT library (C++ bindings)
+ Version: @VERSION@
+ Cflags: -I${includedir}
++Cflags.private: -DLIBMOSQUITTO_STATIC
+ Libs: -L${libdir} -lmosquittopp

--- a/mingw-w64-mosquitto/PKGBUILD
+++ b/mingw-w64-mosquitto/PKGBUILD
@@ -4,7 +4,7 @@ _realname=mosquitto
 pkgbase=mingw-w64-${_realname}
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
 pkgver=2.0.18
-pkgrel=2
+pkgrel=3
 pkgdesc="An Open Source MQTT Broker (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw64' 'ucrt64' 'clang64' 'clangarm64')
@@ -23,10 +23,21 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
              "${MINGW_PACKAGE_PREFIX}-ninja"
              "${MINGW_PACKAGE_PREFIX}-winpthreads"
              "${MINGW_PACKAGE_PREFIX}-docbook-xsl")
-source=("https://mosquitto.org/files/source/mosquitto-${pkgver}.tar.gz"{,.asc})
+source=("https://mosquitto.org/files/source/mosquitto-${pkgver}.tar.gz"{,.asc}
+        '0001-mosquitto-2.0.18-fix-static-libname.patch'
+        '0002-mosquitto-2.0.18-pkgconf-static-cflags.patch')
 sha256sums=('d665fe7d0032881b1371a47f34169ee4edab67903b2cd2b4c083822823f4448a'
-            'SKIP')
+            'SKIP'
+            '153ad2cf3fd8708be469ba2c638e9951ef27841fbd3305b96e5b37781fe5fd09'
+            '288e370e08fcf845a40585b64e78cc79e1063d94468be5aff8b0fbbb0c3122f6')
 validpgpkeys=('A0D6EEA1DCAE49A635A3B2F0779B22DFB3E717B7') # Roger A. Light <roger@atchoo.org>
+
+prepare() {
+  cd "${srcdir}/${_realname}-${pkgver}"
+
+  patch -p1 -i "${srcdir}/0001-mosquitto-2.0.18-fix-static-libname.patch"
+  patch -p1 -i "${srcdir}/0002-mosquitto-2.0.18-pkgconf-static-cflags.patch"
+}
 
 build() {
   mkdir -p "${srcdir}/build-${MSYSTEM}" && cd "${srcdir}/build-${MSYSTEM}"

